lib/internal/Magento/Framework/Filesystem/Driver/StatefulFile.php

```diff
@@ -0,0 +1,1005 @@
+<?php
+/**
+ * Origin filesystem driver
+ *
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Framework\Filesystem\Driver;
+
+use Magento\Framework\Exception\FileSystemException;
+use Magento\Framework\Filesystem\DriverInterface;
+use Magento\Framework\Filesystem\Glob;
+
+/**
+ * Filesystem driver that uses the local filesystem.
+ * Assumed that stat cache is cleanup before test filesystem
+ *
+ * @package Magento\Framework\Filesystem\Driver
+ * @SuppressWarnings(PHPMD.ExcessiveClassComplexity)
+ */
+class StatefulFile implements DriverInterface
```

Comment:
Is it just and old class copied without modification?

Reply:
Mostly yes, except the updates to the clearstatcache. Now it cleanup only requested file (see second params)

Reply:
@kandy ok, then I'll skip parts like new `protected` properties/methods and other code smell.